### ISSUE TITLE
Performance degradation in LTS-14.20

### ISSUE THREAD

**Issue by masterdezign** [lehins/massiv#91]
I have observed about 2.3x speed drop when upgrading from LTS-11.22 to LTS-14.20.

Code example https://github.com/masterdezign/10-days-of-grad/tree/massiv0.4/day4 (commit 56557181414c26801aa88809de824fb944e7a97b)

How to reproduce: 

0. `cd day4`

1. `stack build && time ./run.sh`

2. Apply this diff:

```
diff --git a/day4/stack.yaml b/day4/stack.yaml
index 98949a3..5052771 100644
--- a/day4/stack.yaml
+++ b/day4/stack.yaml
@@ -1,7 +1,7 @@
 # For more information, see: https://github.com/commercialhaskell/stack/blob/release/doc/yaml_configuration.md
 # vim: nospell
 
-resolver: lts-11.22
+resolver: lts-14.20
 
 packages:
 - '.'
```

and run again

`stack build && time ./run.sh`

**Comment by masterdezign** [comment]
Fix:

```
diff --git a/day4/stack.yaml b/day4/stack.yaml
index 98949a3..90c420e 100644
--- a/day4/stack.yaml
+++ b/day4/stack.yaml
@@ -1,7 +1,7 @@
 # For more information, see: https://github.com/commercialhaskell/stack/blob/release/doc/yaml_configuration.md
 # vim: nospell
 
-resolver: lts-11.22
+resolver: nightly-2020-01-21
 
 packages:
 - '.'
```

**Comment by lehins** [comment]
Thank you for reporting this.
For benefit of others, I've also tried the this reproduction of regression in performance and this problem is indeed present when compiled with ghc-8.4 and 8.6 versions. This regressions seems to have been fixed in ghc-8.8, which I also was able to confirm running with `nightly-2020-01-21` resolver.
Just for the record, it doesn't seem to be related to #70
